### PR TITLE
Using new dart_web_audio package to provide web support

### PR DESCRIPTION
**Pull request by luanpotter**
This is a second try on flutter web, using the dart_web_audio that I created, following instructions from the Google/Flutter team.

### CODE REVIEW

**Comment by luanpotter** [comment]
According to my testing, this works on web and does not break mobile.

File: analysis_options.yaml [hunk 1]
```diff
@@ -1,6 +1,10 @@
 # Source of linter options:
 # http://dart-lang.github.io/linter/lints/options/options.html
 
+analyzer:
+  enable-experiment:
+    - extension-methods
+
```
Comment (renancaraujo) [inline analysis_options.yaml]: You dont need this. Just update your SDK constraints on pubspec to >2.6:
```yaml
environment:
  sdk: ">=2.6.0-dev.50.0 <3.0.0"
```

File: pubspec.yaml [hunk 2]
```diff
@@ -13,9 +13,9 @@ flutter:
         pluginClass: AudioplayersPlugin
       macos:
         pluginClass: AudioplayersPlugin
-      # web:
-      #   pluginClass: AudioplayersPlugin
-      #   fileName: audioplayers_web.dart
+      web:
+        pluginClass: AudioplayersPlugin
+        fileName: audioplayers_web.dart
```
Comment (renancaraujo) [inline pubspec.yaml]: Noice

File: analysis_options.yaml (hunk 1 above)
Comment (luanpotter) [reply analysis_options.yaml]: I did that as well. I was "doing things until it worked" and forget to double check which things were actually necessary haha

File: pubspec.yaml (hunk 2 above)
Comment (luanpotter) [reply pubspec.yaml]: Toit

File: lib/audio_cache.dart [hunk 3]
```diff
@@ -120,15 +121,23 @@ class AudioCache {
...
-      file.path,
+      url,
       volume: volume,
       respectSilence: isNotification ?? respectSilence,
       stayAwake: stayAwake,
     );
     return player;
   }
+
+  Future<String> getAbsaoluteUrl(String fileName) async {
```
Comment (erickzanardo) [inline lib/audio_cache.dart]: `getAbsoluteUrl` typo ?

Comment (luanpotter) [reply lib/audio_cache.dart]: Absaolutely :p

**Pull request merged**